staging/src/k8s.io/kubectl/pkg/cmd/util/factory_client_access.go

```diff
@@ -100,46 +117,58 @@ func (f *factoryImpl) RESTClient() (*restclient.RESTClient, error) {
 	if err != nil {
 		return nil, err
 	}
+	httpClient, err := f.ToHTTPClient()
+	if err != nil {
+		return nil, err
+	}
 	setKubernetesDefaults(clientConfig)
-	return restclient.RESTClientFor(clientConfig)
+	return restclient.RESTClientForConfigAndClient(clientConfig, httpClient)
 }
 
 func (f *factoryImpl) ClientForMapping(mapping *meta.RESTMapping) (resource.RESTClient, error) {
-	cfg, err := f.clientGetter.ToRESTConfig()
+	clientConfig, err := f.ToRESTConfig()
```

Comment:
nit: if possible, keep the variable names the same to minimize diff for backports